python/ray/serve/application.py

```diff
@@ -193,6 +201,8 @@ def to_yaml(self, f: Optional[TextIO] = None) -> Optional[str]:
                 yaml.safe_dump(). Returned only if no file pointer is passed in.
         """
 
+        self._convert_to_import_paths()
+
```

Comment:
this is mutating, you probably don't want to do that (else an exception will render the obj in a weird state)?